lib/pure/asynchttpserver.nim

```diff
@@ -315,7 +300,7 @@ proc processRequest(
         await request.respondError(Http413)
         return false
 
-      when (NimMajor, NimMinor) >= (1, 1):
+      if server.stream:
         var remainder = contentLength
         while remainder > 0:
           let readSize = min(remainder, chunkSize)
```

Comment:
I might be missing something, but this whole streaming code looks wrong to me. You are going to be writing all the data into the bodyStream so you are effectively just reimplementing what happens when `stream` isn't set. How does this enable streaming when the code which uses the `Request` only gets it after the `bodyStream` is written to fully?

Reply:
You are right -( please check the new version that use a iterator to read the chunks of data.